src/gc/gc.d

```diff
@@ -1887,6 +1894,59 @@ struct Gcx
         return pool.npages * PAGESIZE;
     }
 
+    /**
+     * Update the thresholds for when to collect the next time
+     */
+    void updateCollectThresholds() nothrow
+    {
+        static float max(float a, float b) nothrow
+        {
+            return a >= b ? a : b;
+        }
+
+        static size_t roundUp(float val, size_t multiple) nothrow
+        {
+            return cast(size_t)(val + multiple - 1 + 0.5) / multiple * multiple;
+        }
+
+        static float zeroDenormal(float val) nothrow
+        {
+            return val + float.min_normal - float.min_normal;
```

Comment:
I'm not sure this really avoids denormals. Even if it is not optimized away, a denormal will be changed to a normal number and then back to a denormal. In this case you don't need to care for denormals because you are dealing with large numbers (the used memory).